modules/seqexec/server/src/main/scala/seqexec/server/gmos/Gmos.scala

```diff
@@ -127,6 +128,11 @@ abstract class Gmos[F[_]: Sync, T<:GmosController.SiteDependentTypes](controller
     } yield new GmosController.GmosConfig[T](configTypes)(cc, dc)
   )
 
+  override def calcStepType(config: Config): Either[SeqexecFailure, StepType] =
+    Gmos.isNodAndShuffle(config).option(NodAndShuffle(instrument).asRight).getOrElse{
+      SequenceConfiguration.calcStepType(config)
```

Comment:
`Gmos.isNodAndShuffle(config).fold(NodAndShuffle(instrument).asRight, SequenceConfiguration.calcStepType(config))`

Reply:
wasn't the prevalent opinion that `fold` on option was bad? @swalker2m

Reply:
This is `fold` on `boolean`.

Reply:
I think @tpolecat might have expressed some concern about `.fold` on `Boolean` since it can be difficult to remember which comes first?  I know it is very unhip but `if` expression seems the most readable to me personally

```scala
  if (Gmos.isNodAndShuffle(config)) NodAndShuffle(instrument).asRight
  else SequenceConfiguration.calcStepType(config)
```

That said I endorse any of these options.